### DISCUSSION TITLE
<style> tag in a page dropped during the build process.

### DISCUSSION THREAD

**Discussion by vivekragunathan** [thuliteio/doks#594] [Q&A]
Hugo supports `<style>` tag in a page. Doks drops the style tag during the build process. Will this be supported? 

The way this is useful is that it makes it easier and cleaner than adding inline styles to HTML elements added to the page. Also, avoids repeating styles inline to each element.

**Comment by h-enk** [comment]
I advise you to minimize the use of inline scripts and styles — for security reasons mainly.

But, if you would like to make use of it (Doks does support it), make sure to use a hash (or nonce), and update Doks' default Content Security Policy accordingly.

See also:
- [CSP: style-src](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src)
- [Security Headers](https://getdoks.org/docs/reference-guides/security/#security-headers)
- [Policy Generator](https://csper.io/generator)
